program_intent_engagement/apps/api/v1/urls.py

```diff
@@ -1,4 +1,15 @@
 """ API v1 URLs. """
+from django.urls import re_path
+
+from program_intent_engagement.apps.api.v1.views import ProgramIntentsView
 
 app_name = 'v1'
-urlpatterns = []
+
+urlpatterns = [
+    re_path(fr'^program_intents/user/<user>/program_uuid/<program_uuid>/reason/<reason>/certainty<certainty\
```

Comment:
the exams endpoint also pulls out most of the handling into separate functions, which can make testing easier because you can test a function which just takes the new program intent and inserts it or not in various scenarios, and then the view just does simple deserialization and needs very little testing